cmake/scripts/osx/ArchSetup.cmake

```diff
@@ -24,6 +24,15 @@ else()
   endif()
 endif()
 
+if(CPU STREQUAL arm64)
+  set(ENABLE_XBMCHELPER OFF)
+else()
+  set(ENABLE_XBMCHELPER ON)
+endif()
+if(ENABLE_XBMCHELPER)
```

Comment:
@fuzzard I think I addressed all your comments, thanks for the review.
The only thing I could not manage to do was to set the `HAS_XBMCHELPER` in the Extratargets.cmake file. It seems that if added there, it's too late for the main application.
Added the check here which I think it should not hurt at all. This should be nuked at some point in the near future anyway.

Reply:
No worries then. However, move the list(APPEND SYSTEM_DEFINES -DHAS_XBMCHELPER) into the above else. No reason to do a second if statement when its known outcome was derived earlier.

Reply:
Sure, did it this way to allow for building for arm64 with xbmchelper support. Doesn't make sense but well..
Will do tomorrow.